docs/examples/pvfleets-qa-pipeline/pvfleets-temperature-qa.py

```diff
@@ -0,0 +1,242 @@
+                                                  marker='o',
+                                                  color="yellow")
+    labels.append("Abnormal")
+if any(zscore_outlier_mask):
+    time_series.loc[zscore_outlier_mask].plot(ls='',
+                                              marker='o',
+                                              color="purple")
+    labels.append("Outlier")
+plt.legend(labels=labels)
+plt.title("Time Series Labeled for Basic Issues")
+plt.xticks(rotation=20)
+plt.xlabel("Date")
+plt.ylabel("Temperature")
+plt.tight_layout()
+plt.show()
+
+# %%
+# Now, let's filter out any of the flagged data from the basic temperature
+# checks (stale or abnormal data). Then we can re-visualize the data
+# post-filtering.
+
+# Filter the time series, taking out all of the issues
+time_series = time_series[~stale_data_mask]
+time_series = time_series[temperature_limit_mask]
+time_series = time_series[~zscore_outlier_mask]
+time_series = time_series.asfreq(data_freq)
+
+# Visualize the time series post-filtering
+time_series.plot(title="Time Series Post-Basic Data Filtering")
+plt.show()
+
+# %%
+# We filter the time series based on its daily completeness score. This
+# filtering scheme requires at least 25% of data to be present for each day to
+# be included. We further require at least 10 consecutive days meeting this
+# 25% threshold to be included.
+
+# Visualize daily data completeness
+data_completeness_score = gaps.completeness_score(time_series)
+
+# Visualize data completeness score as a time series.
+data_completeness_score.plot()
+plt.xlabel("Date")
+plt.ylabel("Daily Completeness Score (Fractional)")
+plt.axhline(y=0.25, color='r', linestyle='-',
+            label='Daily Completeness Cutoff')
+plt.legend()
+plt.tight_layout()
+plt.show()
+
+# Trim the series based on daily completeness score
+trim_series = pvanalytics.quality.gaps.trim_incomplete(
+    time_series,
+    minimum_completeness=.25,
+    freq=data_freq)
+first_valid_date, last_valid_date = \
+    pvanalytics.quality.gaps.start_stop_dates(trim_series)
+time_series = time_series[first_valid_date.tz_convert(time_series.index.tz):
+                          last_valid_date.tz_convert(time_series.index.tz)]
+time_series = time_series.asfreq(data_freq)
+
+# %%
+# Next, we check the time series for any abrupt data shifts. We take the
+# longest continuous part of the time series that is free of data shifts.
+# We use :py:func:`pvanalytics.quality.data_shifts.detect_data_shifts` to
+# detect data shifts in the time series.
+
+# Resample the time series to daily mean
+time_series_daily = time_series.resample('D').mean()
+data_shift_start_date, data_shift_end_date = \
+    ds.get_longest_shift_segment_dates(time_series_daily)
+data_shift_period_length = (data_shift_end_date -
+                            data_shift_start_date).days
+
+# Get the number of shift dates
+data_shift_mask = pvanalytics.quality.data_shifts.detect_data_shifts(
```

Comment:
`ds.detect_data_shifts`

Reply:
Addressed in all scripts in https://github.com/pvlib/pvanalytics/pull/202/commits/cea4fbb93f97e7044f6b50102e1c8bdcf90cd7f4